src/sagas.js

```diff
@@ -163,37 +175,62 @@ function* fetchIcon ({ iconId }) {
 }
 
 function* refreshData (action) {
-  console.log('refreshData(): ', action.payload)
-  if (!action.payload.quiet) {
-    logDebug('refreshData(): not quiet')
+  logDebug('refreshData(): ', action.payload)
+
+  // refresh VMs and remove any that haven't been refreshed
+  const vmsFetchedById = []
+  for (let vmId of Selectors.getVmsFetchedById().toArray()) {
```

Comment:
This can be parallelized by using `all()`.

Reply:
True.  I don't think it'll make much of a difference since I don't see the VMs fetched by ID list getting very long.

I'll revisit.